### PR TITLE
upgrade: Allow granular node upgrade

### PR DESCRIPTION
**Pull request by MaximilianMeister**
### depends on

https://github.com/crowbar/crowbar-core/pull/1056

### CODE REVIEW

File: lib/crowbar/client/command/upgrade/nodes.rb [hunk 1]
```diff
@@ -34,8 +34,16 @@ def execute
...
               when 200
-                say "Successfully triggered the upgrade of the nodes. " \
-                  "Query the upgrade status to follow the process with 'crowbarctl upgrade status'."
+                case args.component
+                when "all"
+                  say "Successfully triggered the upgrade of the nodes. "
+                when "controllers"
+                  say "Successfully triggered the upgrade of the controller nodes. "
+                else
+                  say "Successfully triggered the upgrade of compute node #{args.component}. "
```
Comment (jsuchome) [inline lib/crowbar/client/command/upgrade/nodes.rb]: Do not mention *compute* here. It's true we're only thinking about compute nodes, but better to not limit us for possible changes

**Comment by MaximilianMeister** [comment]
[(5) Allow user to choose which compute nodes to upgrade](https://trello.com/c/IZ8W5tQz/224-5-allow-user-to-choose-which-compute-nodes-to-upgrade)

File: lib/crowbar/client/app/upgrade.rb [hunk 2]
```diff
@@ -184,16 +184,23 @@ def admin
           catch_errors(e)
         end
 
-        desc "nodes",
-          "Trigger the upgrade on all nodes"
+        desc "nodes COMPONENT",
+          "Trigger the node upgrade"
```
Comment (jsuchome) [inline lib/crowbar/client/app/upgrade.rb]: Could you please show the COMPONENTS here as well? (Just the values, it's enough space there)

Comment (MaximilianMeister) [inline lib/crowbar/client/app/upgrade.rb]: done

**Pull request merged**